adapters/rubicon/rubicon_test.go

```diff
@@ -608,6 +608,73 @@ func TestOpenRTBFirstPartyDataPopulating(t *testing.T) {
 	}
 }
 
+func TestPbsHostInfoPopulating(t *testing.T) {
+	bidder := RubiconAdapter{
+		URI:          "url",
+		externalURI:  "externalUrl",
+		XAPIUsername: "username",
+		XAPIPassword: "password",
+	}
+
+	request := &openrtb2.BidRequest{
+		ID: "test-request-id",
+		Imp: []openrtb2.Imp{{
+			ID: "test-imp-id",
+			Banner: &openrtb2.Banner{
+				Format: []openrtb2.Format{
+					{W: 300, H: 250},
+				},
+			},
+			Ext: json.RawMessage(`{
+				"bidder": {
+					"zoneId": 8394,
+					"siteId": 283282,
+					"accountId": 7891,
+					"inventory": {"key1" : "val1"},
+					"visitor": {"key2" : "val2"}
+				}
+			}`),
+		}},
+		App: &openrtb2.App{
+			ID:   "com.test",
+			Name: "testApp",
+		},
+	}
+
+	reqs, _ := bidder.MakeRequests(request, &adapters.ExtraRequestInfo{})
+
+	rubiconReq := &openrtb2.BidRequest{}
+	if err := json.Unmarshal(reqs[0].Body, rubiconReq); err != nil {
+		t.Fatalf("Unexpected error while decoding request: %s", err)
+	}
+
+	var rpImpExt rubiconImpExt
+	if err := json.Unmarshal(rubiconReq.Imp[0].Ext, &rpImpExt); err != nil {
+		t.Fatalf("Error unmarshalling imp.ext: %s", err)
+	}
+
+	var pbsLogin string
+	pbsLogin, err := jsonparser.GetString(rpImpExt.RP.Target, "pbs_login")
+	if err != nil {
+		t.Fatal("Error extracting pbs_login")
+	}
+	assert.Equal(t, pbsLogin, "username", "Unexpected pbs_login value")
+
+	var pbsVersion string
+	pbsVersion, err = jsonparser.GetString(rpImpExt.RP.Target, "pbs_version")
+	if err != nil {
+		t.Fatal("Error extracting pbs_version")
+	}
+	assert.Equal(t, pbsVersion, "", "Unexpected pbs_version value")
```

Comment:
Any ideas on how to actually test version?
In pbs Java we just mock it. But it seems like here nobody tests it. 
I can set `version.Ver` to test value at the beginning of test and defer its reset, but it is a dirty hack.

Reply:
I'm not sure there is any value in adding any additional test logic here. The version is just a string. Checking that it is empty seems sufficient to me since it will predictably always be an empty string in your unit tests.
We also already have test coverage for the version package in `endpoints/version_test.go` so we can feel confident that `Ver` is set correctly.